src/app/vendorEngramsXyzApi/vendorEngramsXyzService.ts

```diff
@@ -0,0 +1,90 @@
+import { VendorDrop, VendorDropType, VendorDropXyz, toVendorDrop } from './vendorDrops';
+import { t } from 'app/i18next-t';
+
+export class VendorEngramsXyzService {
+  refreshInterval: number = 1000 * 60 * 15;
+  cachedResponse: VendorDrop[];
+  lastUpdated = 0;
+  refreshPromise?: Promise<VendorDropXyz[]>;
+
+  handleVendorEngramsErrors(response: Response): Promise<VendorDropXyz[]> {
+    if (response.status !== 200) {
+      throw new Error(t('VendorEngramsXyz.ServiceCallFailed'));
+    }
+
+    return response.json() || [];
+  }
+
+  async dropsNeedRefresh(): Promise<boolean> {
+    const vendorDrops = await this.getAllVendorDrops();
+
+    if (!vendorDrops) {
+      return false;
+    }
+
+    return Boolean(vendorDrops.find((vd) => vd.nextRefresh <= new Date()));
+  }
+
+  cacheExpired(): boolean {
+    if (!this.lastUpdated) {
+      return true;
+    }
+
+    return Date.now() - this.lastUpdated >= this.refreshInterval;
+  }
+
+  vendorEngramsFetch(url: string) {
+    const request = new Request(url, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json'
+      }
+    });
+
+    return Promise.resolve(fetch(request));
+  }
+
+  async getAllVendorDrops(): Promise<VendorDrop[]> {
+    const dropsNeedRefresh = await this.dropsNeedRefresh();
```

Comment:
I'm confused. This awaits `dropsNeedRefresh`, which awaits `getAllVendorDrops`... seems like an infinite loop.